lib/matplotlib/widgets.py

```diff
@@ -2252,23 +2265,24 @@ def direction(self, direction):
         _api.check_in_list(['horizontal', 'vertical'], direction=direction)
         if hasattr(self, '_direction') and direction != self._direction:
             # remove previous artists
-            self._rect.remove()
+            self.selection_artist.remove()
             if self._interactive:
                 self._edge_handles.remove()
-                for artist in self._edge_handles.artists:
-                    self.artists.remove(artist)
             self._direction = direction
             self.new_axes(self.ax)
             if self._interactive:
-                self._setup_edge_handle(self._edge_handles._line_props)
+                line = self._edge_handles.artists[0]
+                line_props = dict(alpha=line.get_alpha(),
```

Comment:
Is there a way to get the properties relevant to `axvline` (the kwargs of `axvline`)? `line.properties()` returns too many properties.

Reply:
Not directly. https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axvline.html#matplotlib.axes.Axes.axvline says

> Valid keyword arguments are Line2D properties, with the exception of 'transform'.

That's because it's a `Line2D`, but we enforce the vertical placement using a transform, so this is not user-settable.

Reply:
Yes, indeed, but there are also other properties (internal properties?), which are not listed in https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axvline.html#matplotlib.axes.Axes.axvline: 

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
line =  ax.axvline(1.5)

props = line.properties()
del props['transform']
line.set(**props)
```
Will give the following error:
```python

  File "/home/eric/Dev/others/matplotlib/lib/matplotlib/artist.py", line 115, in <lambda>
    cls.set = lambda self, **kwargs: Artist.set(self, **kwargs)

  File "/home/eric/Dev/others/matplotlib/lib/matplotlib/artist.py", line 1155, in set
    return self.update(kwargs)

  File "/home/eric/Dev/others/matplotlib/lib/matplotlib/artist.py", line 1060, in update
    raise AttributeError(f"{type(self).__name__!r} object "

AttributeError: 'Line2D' object has no property 'children'
```